baseapp/baseapp.go

```diff
@@ -761,3 +768,30 @@ func (app *BaseApp) runMsgs(ctx sdk.Context, msgs []sdk.Msg, mode runTxMode) (*s
 func makeABCIData(msgResponses []*codectypes.Any) ([]byte, error) {
 	return proto.Marshal(&sdk.TxMsgData{MsgResponses: msgResponses})
 }
+
+func (app *BaseApp) generateFraudProof(storeKeyToSubstoreTraceBuf map[types.StoreKey]*bytes.Buffer) FraudProof {
+	var fraudProof FraudProof
+	fraudProof.stateWitness = make(map[string]StateWitness)
+	cms := app.cms.(*multi.Store)
+
+	for storeKey, subStoreTraceBuf := range storeKeyToSubstoreTraceBuf {
+		keys := cms.GetKVStore(storeKey).(*tracekv.Store).GetAllKeysUsedInTrace(*subStoreTraceBuf)
+
+		substoreSMT := cms.GetSubstoreSMT(storeKey.Name())
+		stateWitness := StateWitness{
+			WitnessData: make([]WitnessData, 0, keys.Len()),
+		}
+		for key := range keys {
+			value := substoreSMT.Get([]byte(key))
+			// Assumption: The keys exist in the SMT because they were traced
+			// TOOD: Investigate default value leaves
```

Comment:
Given https://github.com/celestiaorg/cosmos-sdk/pull/254 removes this and uses `GetProof`, is this still relevant? Would we rather have this PR also change to `GetProof`?

Reply:
I think it's fine to leave this as is and have it fixed in the follow up PR.